File: 1078-remove-outermost-parentheses/1078-remove-outermost-parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
        int counter  = 0 ;
        string ans;
        for(int i=0;i<s.length();i++){
            if(counter>0 && s[i]=='('){
                ans+=s[i];
                counter++;
            } 
            if(s[i]==')' ) counter--;
            if(counter>0 && s[i]==')'){
                ans+=s[i];
                counter--;
            } 
            if(s[i]=='(' ) counter++;
            

        }
        return ans;
        
    }
};

File: 1078-remove-outermost-parentheses/1078-remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        StringBuilder result = new StringBuilder();
        Stack<Character> st= new Stack<>();
        int start = 0;
        for(int i=0;i<s.length();i++){
            char c = s.charAt(i);
            if(c=='(') {
                st.push(c);
            }
            else{
                st.pop();
            }
            if(st.isEmpty()){
                result.append(s,start+1,i);
                start  =  i + 1;
            }
        }
        return result.toString();
    }
}